data.tf

```diff
@@ -0,0 +1,33 @@
+# Copyright 2017, 2021 Oracle Corporation and/or affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+data "oci_core_vcns" "these" {
+  count = var.create_vcn == true ? 0 : 1
```

Comment:
Do we need the count again as its already handled in locals ?

Reply:
I think so : 

```terraform
 vcn_id = var.create_vcn == true ? module.vcn[0].vcn_id : coalesce(var.vcn_id, data.oci_core_vcns.these[0].virtual_networks[0].id)
```